WordPress/src/main/java/org/wordpress/android/util/WPUrlUtils.java

```diff
@@ -0,0 +1,45 @@
+package org.wordpress.android.util;
+
+import java.net.URI;
+import java.net.URL;
+
+public class WPUrlUtils {
+
+    public static synchronized boolean safeToAddWordPressComAuthToken(String url) {
+        return UrlUtils.isHttps(url) && isWordPressCom(url);
+    }
+
+    public static synchronized boolean safeToAddWordPressComAuthToken(URL url) {
+        return UrlUtils.isHttps(url) && isWordPressCom(url);
+    }
+
+    public static synchronized boolean safeToAddWordPressComAuthToken(URI uri) {
+        return UrlUtils.isHttps(uri) && isWordPressCom(uri);
+    }
+
+    public static synchronized boolean isWordPressCom(String url) {
+        return UrlUtils.getHost(url).endsWith(".wordpress.com") || UrlUtils.getHost(url).equals("wordpress.com");
+    }
+
+    public static synchronized boolean isWordPressCom(URL url) {
+        if (url == null) {
+            return false;
+        }
+        return url.getHost().endsWith(".wordpress.com") || url.getHost().equals("wordpress.com");
```

Comment:
Which one of those calls above?

The only class that can return `null` on `getHost()` call is the `URI`, where there is already a null check.